runtime/afl.c

```diff
@@ -113,10 +114,15 @@ CAMLexport value caml_setup_afl(value unit)
   }
   afl_read();
 
+  /* ensure that another module has not already spawned a domain */
+  if (!caml_domain_alone())
+    caml_fatal_error("afl-fuzz: cannot fork with multiple domains running");
```

Comment:
This should call `caml_domain_is_multicore` instead: if your domain is alone but has already spawned a backup thread, `fork` will fail even though `caml_domain_alone` returns true.

Note that this will make `afl.c` depend on `domain.c`, so the linking order must change. Hopefully, it doesn't create a dependency loop.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/747237268f92516c5e1f8c03b3fa4e05f900b4cc